
introduce -Wundef check to catch macro typos
more infos about motivation at https://sourceware.org/glibc/wiki/Wundef

patch 01/04: use always defined MC_HOMEDIR_XDG macro

Signed-off-by: Andreas Mohr <and@gmx.li>

--- a/configure.ac
+++ b/configure.ac
@@ -376,6 +376,8 @@
     AC_DEFINE(MC_HOMEDIR_XDG, 1,
 	[Define to enable XDG standard support])
 else
+    AC_DEFINE(MC_HOMEDIR_XDG, 0,
+	[Define to enable XDG standard support])
     AC_DEFINE_UNQUOTED([MC_USERCONF_DIR], ["$with_homedir"], [Where configs will be placed relative to $HOME])
 fi
 
--- a/lib/mcconfig/paths.c
+++ b/lib/mcconfig/paths.c
@@ -102,7 +102,7 @@
     /* *INDENT-ON* */
 };
 
-#ifdef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG
 static const struct
 {
     char **old_basedir;
@@ -286,7 +286,7 @@
 mc_config_init_config_paths (GError ** mcerror)
 {
     char *dir;
-#ifndef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG == 0
     char *defined_userconf_dir;
 #endif
     mc_return_if_error (mcerror);
@@ -297,7 +297,7 @@
     /* init mc_home and homedir if not yet */
     (void) mc_config_get_home_dir ();
 
-#ifdef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG
     if (mc_home != NULL)
     {
         dir = g_build_filename (mc_home, ".config", (char *) NULL);
@@ -374,7 +374,7 @@
         return;
 
     g_free (mc_config_str);
-#ifdef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG
     g_free (mc_cache_str);
     g_free (mc_data_str);
 #endif /* MC_HOMEDIR_XDG */
@@ -452,7 +452,7 @@
     old_dir = mc_config_get_deprecated_path ();
 
     g_free (mc_config_init_one_config_path (mc_config_str, EDIT_DIR, mcerror));
-#ifdef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG
     g_free (mc_config_init_one_config_path (mc_cache_str, EDIT_DIR, mcerror));
     g_free (mc_config_init_one_config_path (mc_data_str, EDIT_DIR, mcerror));
 #endif /* MC_HOMEDIR_XDG */
@@ -481,7 +481,7 @@
         g_free (old_name);
     }
 
-#ifdef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG
     *msg = g_strdup_printf (_("Your old settings were migrated from %s\n"
                               "to Freedesktop recommended dirs.\n"
                               "To get more info, please visit\n"
--- a/tests/lib/mcconfig/user_configs_path.c
+++ b/tests/lib/mcconfig/user_configs_path.c
@@ -37,7 +37,7 @@
 
 #define HOME_DIR "/home/testuser"
 
-#ifdef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG
 #define CONF_MAIN HOME_DIR PATH_SEP_STR ".config"
 #define CONF_DATA HOME_DIR PATH_SEP_STR ".local" PATH_SEP_STR "share"
 #define CONF_CACHE HOME_DIR PATH_SEP_STR ".cache"
@@ -54,7 +54,7 @@
 setup (void)
 {
     g_setenv ("HOME", HOME_DIR, TRUE);
-#ifdef MC_HOMEDIR_XDG
+#if MC_HOMEDIR_XDG
     g_setenv ("XDG_CONFIG_HOME", CONF_MAIN, TRUE);
     g_setenv ("XDG_DATA_HOME", CONF_DATA, TRUE);
     g_setenv ("XDG_CACHE_HOME", CONF_CACHE, TRUE);
